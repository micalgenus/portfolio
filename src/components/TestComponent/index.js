import React from 'react';
import PropTypes from 'prop-types';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

export default class TestComponent {
  constructor(props) {
    this.props = {
      name: props.name,
      component: props.namcomponente,
      snapshot: props.snapshot === true ? true : false,
      args: props.args || {},
    };
  }

  run = callback => {
    const component = this.props.component;
    const args = this.props.args;

    describe(this.props.name, () => {
      if (this.props.snapshot === true) {
        it('snapshot test', () => {
          const wrapper = shallow(<component {...args} />);
          expect(wrapper).toMatchSnapshot();
        });
      }

      if (typeof callback === 'function') callback.bind(this)();
    });
  };
}

TestComponent.defaultProps = { snapshot: false, args: {} };

TestComponent.propTypes = {
  name: PropTypes.string,
  component: PropTypes.object,
  snapshot: PropTypes.bool,
  args: PropTypes.object,
};

const snapshotTest = (name, component, argv) => {
  if (!name || !component) return null;
  const snapshotTestComponent = new TestComponent({ name: name, component: component, snapshot: true, args: argv });
  snapshotTestComponent.run(null);
};

export { snapshotTest };
