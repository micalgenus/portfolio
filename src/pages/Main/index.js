import React from 'react';
import PageTemplate from '@/Pages';

import CategoryTemplate from '@/Components/CategoryTemplate';
import ItemTemplate from '@/Components/ItemTemplate';

import data from '@/data';

export default class Main extends PageTemplate {
  render() {
    return data.map((v, i) => (
      <CategoryTemplate key={i} category={v.category} description={v.description} links={v.links}>
        {v.items && v.items.map((item, index) => <ItemTemplate key={index} {...item} />)}
      </CategoryTemplate>
    ));
  }
}
