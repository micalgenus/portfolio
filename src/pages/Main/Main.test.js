import { snapshotTest } from '@/TestComponent';
import Main from './Main';

snapshotTest('/Pages/Main', Main);
