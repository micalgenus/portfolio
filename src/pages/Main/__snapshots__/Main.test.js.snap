// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Pages/Main snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "0",
    "nodeType": "class",
    "props": Object {
      "category": "한경수",
      "children": undefined,
      "description": "주로 Node.js를 사용하여 개발을 하고있는 신입 백엔드개발자 입니다. 현재 DevOps에 관심이 많습니다.",
      "links": Array [
        Object {
          "color": "black",
          "href": "mailto:micalgenus@gmail.com",
          "icon": "mail",
        },
        Object {
          "color": "black",
          "href": "https://github.com/micalgenus/",
          "icon": "github",
        },
        Object {
          "color": "blue",
          "href": "https://linkedin.com/in/gyeong-su-han/",
          "icon": "linkedin",
        },
        Object {
          "color": "grey",
          "href": "https://micalgenus.github.io/",
          "icon": "edit",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "0",
      "nodeType": "class",
      "props": Object {
        "category": "한경수",
        "children": undefined,
        "description": "주로 Node.js를 사용하여 개발을 하고있는 신입 백엔드개발자 입니다. 현재 DevOps에 관심이 많습니다.",
        "links": Array [
          Object {
            "color": "black",
            "href": "mailto:micalgenus@gmail.com",
            "icon": "mail",
          },
          Object {
            "color": "black",
            "href": "https://github.com/micalgenus/",
            "icon": "github",
          },
          Object {
            "color": "blue",
            "href": "https://linkedin.com/in/gyeong-su-han/",
            "icon": "linkedin",
          },
          Object {
            "color": "grey",
            "href": "https://micalgenus.github.io/",
            "icon": "edit",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "class",
      "props": Object {
        "category": "Work Experience",
        "children": Array [
          <ItemTemplate
            date="2016.12. ~ 2017.12."
            description="VR 미디어를 사용하여 인테리어 업자와 건물주를 연결시켜 주는 웹/어플리케이션 개발"
            links={
              Array [
                Object {
                  "color": "black",
                  "href": "https://github.com/micalgenus/vgram-web",
                  "icon": "github",
                },
                Object {
                  "color": "black",
                  "href": "https://github.com/micalgenus/vgram-app",
                  "icon": "github",
                },
                Object {
                  "color": "black",
                  "href": "https://github.com/micalgenus/vgram-image-server",
                  "icon": "github",
                },
              ]
            }
            title="모빌리티랩"
          />,
        ],
        "description": undefined,
        "links": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "date": "2016.12. ~ 2017.12.",
            "description": "VR 미디어를 사용하여 인테리어 업자와 건물주를 연결시켜 주는 웹/어플리케이션 개발",
            "links": Array [
              Object {
                "color": "black",
                "href": "https://github.com/micalgenus/vgram-web",
                "icon": "github",
              },
              Object {
                "color": "black",
                "href": "https://github.com/micalgenus/vgram-app",
                "icon": "github",
              },
              Object {
                "color": "black",
                "href": "https://github.com/micalgenus/vgram-image-server",
                "icon": "github",
              },
            ],
            "title": "모빌리티랩",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "2",
      "nodeType": "class",
      "props": Object {
        "category": "Projects",
        "children": Array [
          <ItemTemplate
            description="잉여 음식을 효율적으로 처리하고 소비자들에게 신선한 음식을 제공하는 푸드세일 플랫폼."
            links={
              Array [
                Object {
                  "color": "black",
                  "href": "https://github.com/MubaBot/",
                  "icon": "github",
                },
              ]
            }
            title="푸드세일 O2O 주문 챗봇 - 무바"
          />,
          <ItemTemplate
            description="2017 SW 재능기부 챌린지 지원으로 진행된 프로젝트로 심리상담센터를 운영하시는 소상공인을 위한 홈페이지 및 상담 환경 구축."
            links={
              Array [
                Object {
                  "color": "black",
                  "href": "https://jjrodem.org/",
                  "icon": "home",
                },
              ]
            }
            title="심리상담센터"
          />,
        ],
        "description": undefined,
        "links": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "description": "잉여 음식을 효율적으로 처리하고 소비자들에게 신선한 음식을 제공하는 푸드세일 플랫폼.",
            "links": Array [
              Object {
                "color": "black",
                "href": "https://github.com/MubaBot/",
                "icon": "github",
              },
            ],
            "title": "푸드세일 O2O 주문 챗봇 - 무바",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "description": "2017 SW 재능기부 챌린지 지원으로 진행된 프로젝트로 심리상담센터를 운영하시는 소상공인을 위한 홈페이지 및 상담 환경 구축.",
            "links": Array [
              Object {
                "color": "black",
                "href": "https://jjrodem.org/",
                "icon": "home",
              },
            ],
            "title": "심리상담센터",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "3",
      "nodeType": "class",
      "props": Object {
        "category": "Educations",
        "children": Array [
          <ItemTemplate
            date="2018. 06. ~ 2018. 12."
            links={
              Array [
                Object {
                  "color": "black",
                  "href": "http://swmaestro.kr/",
                  "icon": "home",
                },
              ]
            }
            title="Software Maestro 9th."
          />,
        ],
        "description": undefined,
        "links": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "date": "2018. 06. ~ 2018. 12.",
            "links": Array [
              Object {
                "color": "black",
                "href": "http://swmaestro.kr/",
                "icon": "home",
              },
            ],
            "title": "Software Maestro 9th.",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "4",
      "nodeType": "class",
      "props": Object {
        "category": "Toy Projects",
        "children": Array [
          <ItemTemplate
            description="GCP + Travis-ci + Docker 환경에서 React + DevOps개발 공부"
            links={
              Array [
                Object {
                  "color": "black",
                  "href": "https://portfolio.micalgenus.com/",
                  "icon": "home",
                },
                Object {
                  "color": "black",
                  "href": "https://github.com/micalgenus/portfolio",
                  "icon": "github",
                },
                Object {
                  "color": "blue",
                  "href": "https://hub.docker.com/r/micalgenus/portfolio",
                  "icon": "docker",
                },
              ]
            }
            title="Portfolio"
          />,
        ],
        "description": undefined,
        "links": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "description": "GCP + Travis-ci + Docker 환경에서 React + DevOps개발 공부",
            "links": Array [
              Object {
                "color": "black",
                "href": "https://portfolio.micalgenus.com/",
                "icon": "home",
              },
              Object {
                "color": "black",
                "href": "https://github.com/micalgenus/portfolio",
                "icon": "github",
              },
              Object {
                "color": "blue",
                "href": "https://hub.docker.com/r/micalgenus/portfolio",
                "icon": "docker",
              },
            ],
            "title": "Portfolio",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "5",
      "nodeType": "class",
      "props": Object {
        "category": "Contributes",
        "children": Array [
          <ItemTemplate
            links={
              Array [
                Object {
                  "color": "black",
                  "href": "https://github.com/xpressengine/xe-core/commits?author=micalgenus",
                  "icon": "linkify",
                },
              ]
            }
            title="XpressEngine"
          />,
        ],
        "description": undefined,
        "links": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "links": Array [
              Object {
                "color": "black",
                "href": "https://github.com/xpressengine/xe-core/commits?author=micalgenus",
                "icon": "linkify",
              },
            ],
            "title": "XpressEngine",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "6",
      "nodeType": "class",
      "props": Object {
        "category": "Others",
        "children": Array [
          <ItemTemplate
            links={
              Array [
                Object {
                  "color": "black",
                  "href": "https://micalgenus.github.io/",
                  "icon": "home",
                },
              ]
            }
            title="Github Page Blog"
          />,
        ],
        "description": undefined,
        "links": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "links": Array [
              Object {
                "color": "black",
                "href": "https://micalgenus.github.io/",
                "icon": "home",
              },
            ],
            "title": "Github Page Blog",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
